lightningd/opening_control.c

```diff
@@ -660,12 +660,16 @@ static void channel_config(struct lightningd *ld,
 			   u32 *max_to_self_delay,
 			   struct amount_msat *min_effective_htlc_capacity)
 {
+	bool overflows;
+
 	/* FIXME: depend on feerate. */
 	*max_to_self_delay = ld->config.locktime_max;
 
 	/* Take minimal effective capacity from config min_capacity_sat */
-	amount_msat_from_sat_u64(min_effective_htlc_capacity,
+	overflows = amount_msat_from_sat_u64(min_effective_htlc_capacity,
 		ld->config.min_capacity_sat);
+	if (!overflows)
+		fatal("amount_msat overflow for config.min_capacity_sat");
```

Comment:
This var name is very very confusing!  Perhaps best to just:

```
if (!amount_msat_from_sat_u64(min_effective_htlc_capacity, ld->config.min_capacity_sat))
    fatal().
```

Though I wonder about making min_effective... an msat might be simpler, that's a larger change.

Reply:
Thanks for the hint, all this amount_msat handling seemed to turn off my sense for nice code :D